tests/test_mavconn.py

```diff
@@ -19,8 +34,23 @@ def test_initialization():
     assert handler_test == 'handler3'
     with pytest.raises(KeyError, message="That message name key does not exist!"):
         test_mav.pop_handler('TELEMETRY')
-
     test_mav.clear_handler('TELEMETRY')
     test_mav.clear_handler()
-    assert test_mav._stacks == test_clear  
+    assert test_mav._stacks == test_clear
+
+def test_add_timer_work():
+    test_case = MAVLinkConnection(mavfile)
+    assert threading.active_count() == 1
+    with test_case as m:
+        test_case.add_timer(period, handler)
+        test_case.add_timer(period2, handler2)
+        test_case.add_timer(period3, handler3)
+        assert threading.active_count() == 2
+        with test_case._timers_cv:
+            assert len(test_case._timers) == 3
+            assert test_case._timers[0] <= test_case._timers[1]
+            assert test_case._timers[1] <= test_case._timers[2]
```

Comment:
If you ever find yourself testing private members you are probably doing something wrong.  A test should still pass if you completely change how the internals work.  I suggest testing with handlers and make sure they get called.  You can do this with mocking, or possibly without.